### PR TITLE
add Feature functions

### PR DESCRIPTION
**Pull request by yohancabion**

### CODE REVIEW

File: godal.go [hunk 1]
```diff
@@ -2627,13 +2674,117 @@ func (f *Feature) SetGeometry(geom *Geometry, opts ...SetGeometryOption) error {
 	return cgc.close()
 }
 
+//SetGeometryColumnName set the name of feature first geometry field.
+func (f *Feature) SetGeometryColumnName(name string) {
```
Comment (tbonfort) [inline godal.go:2678]: TODO: return error for incorrect field name

File: godal.go [hunk 2]
```diff
@@ -2627,13 +2674,117 @@ func (f *Feature) SetGeometry(geom *Geometry, opts ...SetGeometryOption) error {
...
+}
+
+//SetFID set feature identifier
+func (f *Feature) SetFID(fid int) {
+	// OGR error returned is always none, so we don't handle it
+	C.OGR_F_SetFID(f.handle, C.GIntBig(fid))
+}
+
+//SetFieldValue set feature's field value
+func (f *Feature) SetFieldValue(field Field, value interface{}) error {
```
Comment (tbonfort) [inline godal.go:2694]: This function needs to be guarded by the standard godal error handler in order to catch warnings/errors raised by incompatible values

File: godal.go [hunk 3]
```diff
@@ -2627,13 +2674,117 @@ func (f *Feature) SetGeometry(geom *Geometry, opts ...SetGeometryOption) error {
...
+		C.OGR_F_SetFieldDateTime(
+			f.handle,
+			C.int(field.index),
+			C.int(timeValue.Year()),
+			C.int(timeValue.Month()),
+			C.int(timeValue.Day()),
+			C.int(timeValue.Hour()),
+			C.int(timeValue.Minute()),
+			C.int(timeValue.Second()),
+			C.int(1), // TODO Time zone is not properly handled
```
Comment (tbonfort) [inline godal.go:2736]: do you plan on handling this TODO?

**Review by tbonfort — CHANGES_REQUESTED** [review]

File: godal.go (hunk 1 above)
Comment (yohancabion) [reply godal.go:2678]: What do you consider as incorrect ? I tried several values (special characters, existing field name...) and OGR_GFld_SetName never raise an error or a warning.

Comment (tbonfort) [reply godal.go:2678]: ok, the error would arise at CreateFeature() time

File: godal.go [hunk 4]
```diff
@@ -2627,13 +2674,117 @@ func (f *Feature) SetGeometry(geom *Geometry, opts ...SetGeometryOption) error {
...
+	cname := C.CString(name)
+	defer C.free(unsafe.Pointer(cname))
+	gfdef := C.OGR_F_GetGeomFieldDefnRef(f.handle, C.int(0))
+	if gfdef != nil {
+		C.OGR_GFld_SetName(gfdef, (*C.char)(unsafe.Pointer(cname)))
+	}
+}
+
+//SetFID set feature identifier
+func (f *Feature) SetFID(fid int) {
```
[the remark this answers is not in the document]
Comment (yohancabion) [reply godal.go]: changed

File: godal.go (hunk 3 above)
Comment (yohancabion) [reply godal.go]: time zones properly handled now

File: godal.go (hunk 2 above)
Comment (yohancabion) [reply godal.go]: handler added

File: godal.go [hunk 5]
```diff
@@ -2733,45 +2747,45 @@ func (f *Feature) SetFieldValue(field Field, value interface{}) error {
...
 		cArray.free()
 	case FTBinary:
 		bytesValue, ok := value.([]byte)
 		if !ok {
 			return errors.New("value for this field must be of type '[]byte'")
 		}
-		C.OGR_F_SetFieldBinary(f.handle, C.int(field.index), C.int(len(bytesValue)), unsafe.Pointer(&bytesValue[0]))
+		C.godalFeatureSetFieldBinary(cgc.cPointer(), f.handle, C.int(field.index), C.int(len(bytesValue)), unsafe.Pointer(&bytesValue[0]))
 	default:
 		return errors.New("setting value is not implemented for this type of field")
```
Comment (tbonfort) [inline godal.go:2785]:
```suggestion
		cgc.close() //avoid resource leak
		return errors.New("setting value is not implemented for this type of field")
```

**Review by tbonfort — APPROVED** [review]

**Pull request merged**